### PR TITLE
Add possibility for a honeypot field

### LINKED ISSUE #212: Add honeypot to forms
| Q | A
| --- | ---
| Bug? | no
| New Feature? | Yes
| Sulu Version | 2.0
| Browser Version | Browser name and version

#### Actual Behavior

At the moment there is nor really useful spam prevention for forms there. Only recaptcha.

#### Possible Solutions
Add a honeypot form element for spammers. This should reduce the amount of form filler spammer.

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #212
| Related issues/PRs | -
| License | MIT

#### What's in this PR?

Add possibility for a honeypot field

#### Why?

To avoid spams

#### Example Usage

```yaml
sulu_form:
    dynamic_honeypot_field: "Honey Pot Field Name"
    dynamic_honeypot_strategy: spam # no_save, no_email, spam
```

#### To Do

- [ ] Update CHANGELOG.md
- [x] Create documentation

### CODE REVIEW

**Comment by alexander-schranz** [comment]
ping @wachterjohannes

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -97,6 +98,15 @@ public function getConfigTreeBuilder()
                 ->normalizeKeys(false)
                 ->prototype('scalar')->end()->defaultValue([])
             ->end()
+            ->scalarNode('dynamic_honeypot_field')->defaultValue(null)->end()
```
Comment (wachterjohannes) [inline DependencyInjection/Configuration.php]: why not

```
sulu_form:
   honeypot:
       name: ...
       type: ...
```

File: Form/Type/DynamicFormType.php [hunk 2]
```diff
@@ -149,6 +157,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
         ]);
 
+        if ($this->honeyPotField) {
+            $builder->add(
+                str_replace(' ', '_', strtolower($this->honeyPotField)),
+                EmailType::class,
+                [
+                    'label' => $this->honeyPotField,
+                    'mapped' => false,
+                    'block_prefix' => 'honeypot',
```
Comment (alexander-schranz) [inline Form/Type/DynamicFormType.php]: should also be `required` => false

Comment (almare) [reply Form/Type/DynamicFormType.php]: right! I just want to add this right now, but I saw you already did it :-).

File: Resources/config/services.xml [hunk 3]
```diff
@@ -13,7 +13,8 @@
             <argument type="service" id="twig" />
             <argument type="service" id="event_dispatcher" />
             <argument type="service" id="sulu_media.media_manager" />
-            <argument type="service" id="logger" />
+            <argument>%sulu_form.dynamic_honeypot_strategy%%</argument>
```
Comment (almare) [inline Resources/config/services.xml]: Here is still the typo 2 x %% are wrong

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (alexander-schranz) [reply DependencyInjection/Configuration.php]: Fixed

**Pull request merged**